libs/libvtrutil/src/vtr_geometry.h

```diff
@@ -75,9 +75,10 @@ class Point {
 template<class T>
 class Rect {
   public: //Constructors
+    Rect();
     Rect(T left_val, T bottom_val, T right_val, T top_val);
     Rect(Point<T> bottom_left_val, Point<T> top_right_val);
-    Rect();
+    Rect(Point<T> bottom_left_val, T size);
```

Comment:
This is useful to add points on an integer grid (size = 1), so that `Rect(p, 1).contains(p) == true`, so `size` is the size of the point.

I'm okay with making it width/height, defaulting the size to 1, or doing all of these with overloads.

Just using the smallest size that satisfies the above property (1 for integers, and epsilon for floating point) is probably the most reasonable, and stating the property in comments and adding tests.

Reply:
Using epsilon for floats won't always work, so I think this constructor should just be enabled for integers.

Reply:
I've created a constructor from a single vtr::Point<T> where T is an integral type.